src/parser.lisp

```diff
@@ -1,3 +1,90 @@
 ;;;; Parser of scene files (.obj).
 
 (in-package #:cl-raster/parser)
+
+(defun raw-line-to-tokens (line)
+  (let ((line-no-comments (str:trim
+                            (str:substring
+                              0
+                              (position #\# line)
+                              line))))
+    (str:split " " line-no-comments :omit-nulls T)))
+
+(defun parse-mtl-file (filename)
+  (let ((materials (make-hash-table))
+        (current-mtl NIL))
+    (with-open-file (in-stream filename)
+      (loop for line = (read-line in-stream NIL)
+            while line
+            for tokens = (raw-line-to-tokens line)
+            when tokens
+            do (when (eq (first tokens) "newmtl")
+                 (setf (gethash (second tokens) materials) (scene:make-mtl)
+                       current-mtl (second tokens)))
```

Comment:
Maybe set to `current-mtl` the `mtl` object itself instead of its name?
```lisp
(setf current-mtl (scene:make-mtl)
      (gethash ...) current-mtl)
```

Then you won't need a lookup every time when you set a property of the `mtl` object.